From 54aeb6b955a8a5a2ad8770b9c93b8a93f63eca20 Mon Sep 17 00:00:00 2001
From: Andreas Dumberger <andreas.dumberger@tq-group.com>
Date: Thu, 18 Mar 2021 18:40:18 +0100
Subject: [PATCH] tqmarzg2x: add fix from Renesas for accessing SPI-NOR

This Patch was originally sent from FAE P. Furtner on 2021-03-18 by email.
Without this patch, only after booting from eMMC, Linux could not access
the SPI-NOR memory.

Signed-off-by: Andreas Dumberger <andreas.dumberger@tq-group.com>
---
 drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c | 7 +++++++
 drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c   | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c b/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c
index 8c1de5402..24a414b7a 100644
--- a/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c
+++ b/drivers/renesas/rzg/pfc/G2H/pfc_init_h3_v2.c
@@ -568,10 +568,17 @@
 #define MOD_SEL2_VIN4_A		((uint32_t)0U << 0U)
 #define MOD_SEL2_VIN4_B		((uint32_t)1U << 0U)
 
+#define	SEC_CONF			(0xEE2000B8)
+
 static void pfc_reg_write(uint32_t addr, uint32_t data)
 {
 	mmio_write_32(PFC_PMMR, ~data);
 	mmio_write_32((uintptr_t)addr, data);
+	/*
+	 * Initialize Secure Configuration Register to access QSPI
+	 * when eMMC Boot is performed.
+	 */
+	mmio_write_32(SEC_CONF, 0x0155);
 }
 
 void pfc_init_h3_v2(void)
diff --git a/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c b/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c
index 53a76989e..ebea3c20d 100644
--- a/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c
+++ b/drivers/renesas/rzg/pfc/G2N/pfc_init_m3n.c
@@ -570,10 +570,17 @@
 #define MOD_SEL2_VIN4_A		((uint32_t)0U << 0U)
 #define MOD_SEL2_VIN4_B		((uint32_t)1U << 0U)
 
+#define	SEC_CONF			(0xEE2000B8)
+
 static void pfc_reg_write(uint32_t addr, uint32_t data)
 {
 	mmio_write_32(PFC_PMMR, ~data);
 	mmio_write_32((uintptr_t)addr, data);
+	/*
+	 * Initialize Secure Configuration Register to access QSPI
+	 * when eMMC Boot is performed.
+	 */
+	mmio_write_32(SEC_CONF, 0x0155);
 }
 
 void pfc_init_m3n(void)
-- 
2.25.1

